models/categories-model.js

```diff
@@ -8,12 +8,18 @@ exports.readCategories = () => {
 };
 
 exports.readReviewsById = (review_id) => {
-
-    return db.query(`SELECT * FROM reviews WHERE review_id = $1`, [review_id]).then((result) => {
-      if (result.rows.length === 0) {
+  return Promise.all([
+  db.query(`SELECT * FROM reviews WHERE review_id = $1`, [review_id]),
+  db.query(`SELECT * FROM comments WHERE review_id = $1`, [review_id])])
+  .then(([review, comments]) => {
+    
+      if (review.rows.length === 0) {
         return Promise.reject({status: 404, msg: 'ID does not exist'})
       }
-        return result.rows[0]
+
+       review.rows[0].comment_count = comments.rows.length
+       
+       return review.rows[0]
```

Comment:
Id prefer we did this as one SQL statement. We can do this altogether in a single query using the COUNT() function available in SQL, have a look at the notes here on how to use the COUNT() with a GROUP BY and a JOIN
https://notes.northcoders.com/courses/js-back-end/sql-relationships

Reply:
refactoring applied